frontend/src/components/VExternalSearch/VExternalSearchForm.vue

```diff
@@ -36,72 +36,71 @@
       <template #query>{{ searchTerm }}</template>
     </i18n>
 
-    <VButton
-      id="external-sources-button"
-      ref="triggerRef"
-      :pressed="triggerA11yProps['aria-expanded']"
-      aria-haspopup="dialog"
-      :aria-controls="
-        isMd ? 'external-sources-popover' : 'external-sources-modal'
-      "
-      variant="dropdown-label"
-      size="disabled"
-      class="caption-regular ms-2 min-w-max gap-1 px-3 py-1 pe-1 text-dark-charcoal focus-visible:border-tx"
-      @click="onTriggerClick"
-      >{{ $t("externalSources.button").toString()
-      }}<VIcon
-        class="text-dark-charcoal-40"
-        :class="{ 'text-white': triggerA11yProps['aria-expanded'] }"
-        name="caret-down"
-      />
-    </VButton>
-    <template v-if="triggerElement">
-      <Component
-        :is="isMd ? 'VPopoverContent' : 'VModalContent'"
-        :id="isMd ? 'external-sources-popover' : 'external-sources-modal'"
-        :aria-labelledby="'external-sources-button'"
-        :hide="closeDialog"
-        :trigger-element="triggerElement"
-        :visible="isVisible"
-        :z-index="isMd ? 'popover' : 'modal'"
-        :variant="isMd ? undefined : 'centered'"
-      >
-        <VExternalSourceList
-          class="flex flex-col"
-          :search-term="searchTerm"
-          @close="closeDialog"
-        />
-      </Component>
-    </template>
+    <VModal
+      variant="centered"
+      :hide-on-click-outside="true"
+      labelled-by="external-sources-button"
+      @open="handleModalOpen"
+    >
+      <template #trigger="triggerA11yProps">
+        <VButton
+          id="external-sources-button"
+          v-bind="triggerA11yProps"
+          aria-controls="external-sources-modal"
+          variant="dropdown-label"
+          size="disabled"
+          class="caption-regular ms-2 min-w-max gap-1 px-3 py-1 pe-1 text-dark-charcoal focus-visible:border-tx"
+          >{{ `${$t("externalSources.button")}`
+          }}<VIcon
+            class="text-dark-charcoal-40"
+            :class="{ 'text-white': triggerA11yProps['aria-expanded'] }"
+            name="caret-down"
+          />
+        </VButton>
+      </template>
+      <template #top-bar="{ close }">
+        <header class="flex items-center justify-between pe-4 ps-7 pt-4">
+          <h2 class="heading-6" tabindex="-1">
+            {{ $t("externalSources.title") }}
+          </h2>
+          <VCloseButton
+            size="small"
+            icon-size="medium"
+            variant="filled-white"
+            :label="$t('modal.close')"
+            @close="close"
+          />
+        </header>
+      </template>
+      <VExternalSourceList class="flex flex-col" :search-term="searchTerm" />
```

Comment:
Icon size of external link item needs to be like [VIconButton](https://docs.openverse.org/storybook/?path=/story/components-viconbutton--default-story&args=size:small) in `size: small`, but not interactable. This is for consistenty with the new popover item requested in #2839.

Reply:
Sorry for not explaining it correctly.

The popover item in #2839 has an optional icon on the right side that in some cases you can interact with it. Here is an example.

![recent searches item interaction](https://github.com/WordPress/openverse/assets/895819/fd95e308-c2c7-4409-a1af-3d2105a829fe)

Although the above has not been implemented yet, the new style was thought to allow this internal action.

When the action is not needed, like in this case where `↗` only indicates a link type, the icon area is the same (`32px` for the container, `24px` as icon size) to keep consistency across all popover items. Here is a screenshot of popover item in wireframe.

<img width="500" alt="Outline version of Popover item component" src="https://github.com/WordPress/openverse/assets/895819/618d4a14-4e22-4b7c-bcd4-2d07ff7c79cc">

The smallest box area is the default icon size of `24px`, and the container is like `VIconButton` in `32px (size: small)`.

Reply:
At first when I read this note, I thought maybe we should create a separate component in code. However, because they have different functions (the external icon is a "decoration" that has no functionality and the remove "cross" icon is actually an icon button that should have some click handlers) - I think it's best to leave the "external icon" here as VIcon. I set the icon size to 5, which makes the icon itself around 16px high - just like in the mockups. I think the size of the icon now matches the designs. Could you please confirm, @fcoveram ?

Reply:
> because they have different functions (the external icon is a "decoration" that has no functionality and the remove "cross" icon is actually an icon button that should have some click handlers) - I think it's best to leave the "external icon" here as VIcon

That makes sense to me. But VIcon needs to be `size: 6 (24px)` and placed inside a `32px` box to look balanced. Without the 32px container, it looks broken.

Here is a design test where the color-filled area is the VIcon in size 6, and the border-only area is the 32px container.

![Diagram of popover item. Left, icon with a container. Right, icon without container](https://github.com/WordPress/openverse/assets/895819/c8bb51ec-1e63-4918-b880-c97b87c9fbec)

I reviewed the PR's last changes and still see the icon in 20px. Here is a gif of what I see.

![popover item - chrome inspector](https://github.com/WordPress/openverse/assets/895819/490884cb-a375-46d1-bf89-ed010e75c316)

Reply:
VIcon in size 8 is 32px element that scales up the icon, whereas the above refers to a container area to balance spaces between popover items while keeping the icon in 24px.